pkg/client/client.go

```diff
@@ -244,6 +247,11 @@ func WithReceiptTimeout(timeout time.Duration) CreateOption {
 
 // Create creates a new table on the Tableland.
 func (c *Client) Create(ctx context.Context, schema string, opts ...CreateOption) (TableID, string, error) {
+	if c.tblContract == nil {
+		return TableID{}, "", errors.New(
+			"clinet created with no provider, must provide an Infura API key, Alchemy API key, or an ETH backend",
+		)
+	}
```

Comment:
For calls that require a provider, we check and error if necessary. Here and below as well.